dynamic-property-api/src/main/java/ru/fix/dynamic/property/api/DynamicPropertySource.java

```diff
@@ -0,0 +1,99 @@
+package ru.fix.dynamic.property.api;
+
+import java.util.Map;
+import java.util.Properties;
+
+public interface DynamicPropertySource extends AutoCloseable {
+
+    /**
+     * Returns property value for specified key in required type.
+     * <p>
+     * There are no guarantees of accuracy. This is merely the most recent view
+     * of the data.
+     * </p>
+     * @param key  property name
+     * @param type property type
+     * @return property value, converted to the required type or {@code null} if
+     * there is no such property
+     * @throws IllegalArgumentException if conversation are not supported
+     * @throws NumberFormatException    if property value are not numeric and {@link Integer} or
+     *                                  {@link Long} type are specified
+     */
+    <T> T getProperty(String key, Class<T> type);
+
+    /**
+     * Returns property value for specified key or default.
+     * <p>
+     * There are no guarantees of accuracy. This is merely the most recent view
+     * of the data.
+     * </p>
+     *
+     * @param key          property name
+     * @param type         property type
+     * @param defaultValue default value
+     * @return property value, converted to the required type or
+     * {@code defaultValue} if there is no such property
+     */
+    <T> T getProperty(String key, Class<T> type, T defaultValue);
+
+    /**
+     * Reads up all existing properties
+     *
+     * @return properties with values
+     */
+    Properties getAllProperties() throws Exception;
+
+    /**
+     * Read subtree of properties with optional root node.
+     *
+     * @param root node of subtree, if {@code null} then default config location will be used.
+     * @return key-value entries
+     */
+    Map<String, String> getAllSubtreeProperties(String root) throws Exception;
+
+    /**
+     * Uploads initial properties.
+     *
+     * @param propertiesPath initial properties
+     * @return actual set of properties
+     */
+    Properties uploadInitialProperties(String propertiesPath) throws Exception;
```

Comment:
if we will folow the rule "all default properties should have default values"  then uploadInitialProperties is not neccessary. But i can imagine that it will be useful in the future